### PR TITLE
trie: polish commit function

### PR DESCRIPTION
**Pull request by rjl493456442**
This PR polishes the commit function of stacktrie a bit.

### CODE REVIEW

File: trie/stacktrie.go [hunk 1]
```diff
@@ -391,14 +394,18 @@ func (st *StackTrie) Hash() (h common.Hash) {
...
 
-// Commit will commit the current node to database db
-func (st *StackTrie) Commit(db ethdb.KeyValueStore) common.Hash {
-	oldDb := st.db
-	st.db = db
-	defer func() {
-		st.db = oldDb
-	}()
+// Commit will firstly hash the entrie trie if it's still not hashed
+// and then commit all nodes to the assoicated database. Actually most
```
Comment (gballet) [inline trie/stacktrie.go:398]: typo: associated

File: trie/stacktrie.go [hunk 2]
```diff
@@ -391,14 +394,18 @@ func (st *StackTrie) Hash() (h common.Hash) {
...
+// of the trie nodes MAY have been committed already. The main purpose
+// here is to commit the root node.
+//
+// The assoiciated database is expected, otherwise the whole commit
```
Comment (gballet) [inline trie/stacktrie.go:402]: typo: associated

**Review by gballet — APPROVED** [review]
LGTM, just caught a couple typos.

**Review by holiman — APPROVED** [review]
LGTM

**Pull request merged**